cmake/API.cmake

```diff
@@ -138,367 +138,294 @@ function(add_fprime_subdirectory FP_SOURCE_DIR)
 endfunction(add_fprime_subdirectory)
 
 ####
-# Function `register_fprime_module`:
-#
-# Registers a module using the fprime build system. This comes with dependency management and fprime
-# autocoding capabilities. The caller should first set two variables before calling this function to define the
-# autocoding and source inputs, and (optionally) any non-standard link dependencies.
+# Function `fprime_attach_custom_targets`:
 #
-# Required variables (defined in calling scope):
+# Attaches custom fprime targets (cmake/targets) and their associated autocoding to the supplied build
+# target. This is done automatically by the `register_fprime_*` family of functions and provides deferred
+# target setup for use with `fprime_add_*_build_target` family functions.
 #
-# - **SOURCE_FILES:** cmake list of input source files. Place any "*.fpp", "*.c", "*.cpp"
-#   etc files here. This list will be split into autocoder inputs, and hand-coded sources based on the name/type.
+# **BUILD_TARGET_NAME:** name of build target to attach targets and autocoding to
 #
-# **i.e.:**
-# ```
-# set(SOURCE_FILES
-#     MyComponent.fpp
-#     SomeFile.cpp
-#     MyComponentImpl.cpp)
-# ```
-# - **MOD_DEPS:** (optional) cmake list of extra link dependencies. This is optional, and only
-#   needed if non-standard link dependencies are used, or if a dependency cannot be inferred from the include graph of
-#   the autocoder inputs to the module. If not set or supplied, only fprime inferable dependencies will be available.
-#   Link flags like "-lpthread" can be added here as well. Do NOT supply executable targets in MOD_DEPS. See:
-#   `register_fprime_executable` for alternatives.
-#
-# **i.e.:**
-# ```
-# set(MOD_DEPS
-#     Os
-#     Module1
-#     Module2
-#     -lpthread)
-# ```
-#
-# ### Standard `add_fprime_module` Example ###
-#
-# Standard modules don't require extra modules, and define both autocoder inputs and standard source
-# files. Thus, only the SOURCE_FILE variable needs to be set and then the register call can be made.
-# This is the only required lines in a module CMakeLists.txt.
-#
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp
-#     SomeFile.cpp
-#     MyComponentImpl.cpp)
-#
-# register_fprime_module()
-# ```
+####
+function(fprime_attach_custom_targets BUILD_TARGET_NAME)
+    setup_module_targets("${BUILD_TARGET_NAME}")
+endfunction()
+
+####
+# Function `register_fprime_library`:
 #
-# ### Non-Autocoded and Autocode-Only Modules Example ###
+# Registers a library using the fprime build system. This comes with dependency management and fprime
```

Comment:
What's the rationale for moving away from `module` and into the `library` terminology? Is it because a deployment is also a module?

Also if names are to be changed, should these eventually be `fprime_register_*` instead of `register_fprime_*` ?

Reply:
Yes, module is more overloaded and library is clear to map to a library artifact.

Maybe we should rename, but also i don't have a deep drive to do so.